ims/einstein/operations.py

```diff
@@ -217,11 +217,16 @@ def shutdown(self):
 
     # Provisions from HaaS and Boots the given node with given image
     @log
-    def provision(self, node_name, img_name, network, nic):
+    def provision(self, node_name, img_name, network, nic, hil_flag):
         try:
+            hil_bool = bool(hil_flag == "True")
```

Comment:
Could you please spend a bit more time in figuring out why it is being passed as string?

If it is due to rpc, then try going through pyro4 lib if there is something which wont convert to string.